pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/logicaltable/BaseLogicalTableIntegrationTest.java

```diff
@@ -128,28 +129,40 @@ public void setUp()
       TableConfig offlineTableConfig = createOfflineTableConfig(tableName);
       addTableConfig(offlineTableConfig);
 
-      List<File> offlineAvroFiles = new ArrayList<>(numSegmentsPerTable);
-      for (int i = index; i < index + numSegmentsPerTable; i++) {
-        offlineAvroFiles.add(avroFiles.get(i));
-      }
-      index += numSegmentsPerTable;
-
       // Create and upload segments
-      ClusterIntegrationTestUtils.buildSegmentsFromAvro(offlineAvroFiles, offlineTableConfig, schema, 0, _segmentDir,
+      ClusterIntegrationTestUtils.buildSegmentsFromAvro(avroFilesForTable, offlineTableConfig, schema, 0, _segmentDir,
           tarDir);
       uploadSegments(tableName, tarDir);
     }
 
+    // create realtime table
+    Map<String, List<File>> realtimeTableDataFiles = getRealtimeTableDataFiles();
+    for (Map.Entry<String, List<File>> entry : realtimeTableDataFiles.entrySet()) {
+      String tableName = entry.getKey();
+      List<File> avroFilesForTable = entry.getValue();
+      // create and upload the schema and table config
+      Schema schema = createSchema(getSchemaFileName());
+      schema.setSchemaName(tableName);
+      addSchema(schema);
+
+      TableConfig realtimeTableConfig = createRealtimeTableConfig(avroFilesForTable.get(0));
+      realtimeTableConfig.setTableName(tableName);
+      addTableConfig(realtimeTableConfig);
+
+      // push avro files into kafka
+      pushAvroIntoKafka(avroFilesForTable);
```

Comment:
How does this function decide the topic name ?

Reply:
Default implementation from `BaseClusterIntegrationTest`.

```
  protected String getKafkaTopic() {
    return getClass().getSimpleName();
  }
```

Reply:
All the data will go into the same topic ? For multiple realtime tables there should be multiple kafka topics. Else all the realtime tables will have the same data.

Reply:
We do not have tests with multiple realtime table at the moment. That is what we agreed upon yesterday.

Reply:
ah right - nvm.